### ISSUE TITLE
`ng new game-2d` fails

### ISSUE THREAD

**Issue by leegee** [angular/angular-cli#13885]
# 🐞 Bug report

### Command (mark with an `x`)
```
- [x ] new
- [ ] build
- [ ] serve
- [ ] test
- [ ] e2e
- [ ] generate
- [ ] add
- [ ] update
- [ ] lint
- [ ] xi18n
- [ ] run
- [ ] config
- [ ] help
- [ ] version
- [ ] doc
```

### Is this a regression?

?

### Description

Apparently can't create a new project with a hyphen in the name.

```
leege@studio MINGW64 /l/src/game-2d
$ npm install -g @angular/cli
C:\Users\leege\AppData\Roaming\npm\ng -> C:\Users\leege\AppData\Roaming\npm\node_modules\@angular\cli\bin\ng
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.7 (node_modules\@angular\cli\node_modules\fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.7: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"x64"})

+ @angular/cli@7.3.5
added 110 packages from 91 contributors, removed 61 packages and updated 27 packages in 17.326s

leege@studio MINGW64 /l/src/game-2d
$

leege@studio MINGW64 /l/src/game-2d
$ ng new game-2d
? Would you like to add Angular routing? Yes
? Which stylesheet format would you like to use? CSS    (.css )
Schematic input does not validate against the Schema: {"name":"game-2d","version":"7.3.5","routing":true,"style":"css"}
Errors:

  Data path ".name" should match format "html-selector".

leege@studio MINGW64 /l/src/game-2d
$ ng new game2d
? Would you like to add Angular routing? Yes
? Which stylesheet format would you like to use? CSS    (.css )
CREATE game2d/angular.json (3807 bytes)
CREATE game2d/package.json (1305 bytes)
CREATE game2d/README.md (1023 bytes)...
...etc...
```

## 🔬 Minimal Reproduction

```ng new game-2d```

## 🔥 Exception or Error
As above:

<pre><code>
Schematic input does not validate against the Schema: {"name":"game-2d","version":"7.3.5","routing":true,"style":"css"}
Errors:

  Data path ".name" should match format "html-selector".
</code></pre>

## 🌍 Your Environment
<pre><code>
Angular CLI: 7.3.5
Node: 11.9.0
OS: win32 x64
Angular:
...

Package                      Version
------------------------------------------------------
@angular-devkit/architect    0.13.5
@angular-devkit/core         7.3.5
@angular-devkit/schematics   7.3.5
@schematics/angular          7.3.5
@schematics/update           0.13.5
rxjs                         6.3.3
typescript                   3.2.4
</code></pre>

**Comment by alan-agius4** [comment]
Hi, the issue here is the number as html selectors should not contain digits.

Seeing specs here: https://html.spec.whatwg.org/multipage/custom-elements.html#valid-custom-element-name

**Comment by leegee** [comment]
Perhaps I misreading but the spec to which you point reads as below, where
PCENChar can contain any character in the range [0-9]:

name must match the PotentialCustomElementName
<https://html.spec.whatwg.org/multipage/custom-elements.html#prod-potentialcustomelementname>
 production:
PotentialCustomElementName ::=[a-z] (PCENChar
<https://html.spec.whatwg.org/multipage/custom-elements.html#prod-pcenchar>)*
'-' (PCENChar
<https://html.spec.whatwg.org/multipage/custom-elements.html#prod-pcenchar>
)*PCENChar ::="-" | "." | [0-9] | "_" | [a-z] | #xB7 | [#xC0-#xD6] |
[#xD8-#xF6] | [#xF8-#x37D] | [#x37F-#x1FFF] | [#x200C-#x200D] |
[#x203F-#x2040] | [#x2070-#x218F] | [#x2C00-#x2FEF] | [#x3001-#xD7FF] |
[#xF900-#xFDCF] | [#xFDF0-#xFFFD] | [#x10000-#xEFFFF]

**Comment by leegee** [comment]
Should I create a new issue with reference to the specs, or have I misunderstood?

**Comment by alan-agius4** [comment]
Re-opened as your clarification seems a corrrct. 

Thank you.

**Comment by alan-agius4** [comment]
Hi again, actually, this was fixed in this https://github.com/angular/angular-cli/pull/13837

See this spec https://github.com/angular/angular-cli/pull/13837/files#diff-867e0bcfb9109ac41d6c72f11fc039a0R46

Thanks

**Comment by HugoCapocci** [comment]
I have the same issue with v7.3.8 ; Didnt' have any trouble with 7.3.0 
Why am I forced to use a BETA version of the next MAJOR version in order to have a fix? Why do not prepare 7.3.9 patch?
